File: Sample/AgricultureModel/Solo.m
%
% This script is used to execute the model by iteself (i.e. not via the SSW).
%

clear;
clear global;

% This is used by the placeholder scripts to keep track of the simulation
% time. It is not accessed by the crop choice model directly, instead it is
% accessed via the sswGetCurrentTime function.
global globalYear;

% Include the folder of placeholder scripts since we're running stand-
% alone (not linked). When running via OpenMI, the ssw functions will be
% provided automatically.
addpath('Solo');

% Tell the crop choice model to prepare for a simulation.
sswInitialize();

% Tell the crop choice model to simulate each year.
for globalYear=2011:2040
    sswPerformTimeStep();
end;

% Tell the crop choice model the simulation is over.
sswFinish();

File: Sample/WeatherModel/Solo.py
#
# This script is used to execute the model by iteself (i.e. not via the SSW).
#

import sys
import os
sys.path.append(r"C:\Python27\Lib")
sys.path.append(os.path.abspath('Solo'))
from Model import ModelClass

def main():
        model = ModelClass()
        
        model.sswInitialize()

        for year in range(2011,2041):
                model.globalYear = year
                model.sswPerformTimeStep()

        model.sswFinish()
        
        pass

if __name__ == '__main__':
        main()
